### ISSUE TITLE
Define API

### ISSUE THREAD

**Issue by Notgnoshi** [Notgnoshi/research#66]
* Tags should be implicit.
* prompt (be able to pick a random one, or specify)
* number of haiku
* temperature
* repetition_penalty
* k, p
* seed
* Return JSON with some metadata about generated haiku? Should be optional.

**Comment by Notgnoshi** [comment]
# Generation ` /generate/`

`/generate?prompt=...&seed=...&number=...&temperature=...&k=...&p=...&max_tokens=...`

* `prompt` - _str_ - The word, or words, to begin the haiku with.
* `seed` - _uint32_t_ - Random seed for reproducible results. Need to bounds check?
* `number` - _int_ - The number of haiku to generate. Should _definitely_ enforce an upper bound.
* `temperature` - _float_ Should bounds check. Default to 1.0
* `k` - _int_ - Lookup documentation
* `p` - _float_ - Lookup documentation
* `max_tokens` - _int_ - Max number of tokens to generate. Generation will still stop when it hits the `$` character.

JSON response with
* prompt
* seed
* haiku (obviously)
* temperature, k, p?
* colors?
* syllables?
* ~most similar haiku from the training set?~

Generated haiku should be stored. This should be atomic?

Is there a way to know how overwhelmed the system is? Return an appropriate HTTP response?

# Generated Data Access `/generated/`

Try to make these fast and efficient.

* `/generated/n - Return the `n`th generated haiku.
* `/generated/` - Return a random generated haiku. Redirect to `/generated/n`

# Training

Don't support, because there's no way in hell I'm letting the general public run 4+hr CPU-bound jobs on my desktop. I also don't want to worry about training concurrently?

# Training Data Access `/data/`

Find a way to link to the training set in the API docs.
That way I don't get 55k requests from the same user in the span of a few seconds.

* `/data/n` - Return the `n`th training set haiku.
* `/data/` - Return a random training set haiku. Redirect to `/data/n`.
